packages/knit-gateway/src/schema.ts

```diff
@@ -13,64 +13,101 @@
 // limitations under the License.
 
 import {
-  type MaskField,
   Schema,
-  type Schema_Field,
-  type Schema_Field_Type,
   Schema_Field_Type_ScalarType,
 } from "@buf/bufbuild_knit.bufbuild_es/buf/knit/gateway/v1alpha1/knit_pb.js";
 import { Code, ConnectError } from "@bufbuild/connect";
-import {
-  type MessageType,
-  type FieldInfo,
-  ScalarType,
-  type EnumType,
-  type PlainMessage,
-} from "@bufbuild/protobuf";
+import { ScalarType } from "@bufbuild/protobuf";
 import { wktSet } from "./wkt.js";
+import type {
+  MaskField,
+  Schema_Field,
+  Schema_Field_Type,
+} from "@buf/bufbuild_knit.bufbuild_es/buf/knit/gateway/v1alpha1/knit_pb.js";
+import type {
+  MessageType,
+  FieldInfo,
+  EnumType,
+  PlainMessage,
+  AnyMessage,
+} from "@bufbuild/protobuf";
+import type { Gateway, Relation } from "./gateway.js";
+
+// Instead of creating new types which can be tedious, we use declaration merging to
+// add the fields we need to the existing types. These fields are ignored by
+// `@bufbuild/protobuf`.
```

Comment:
The problem is recursion, with the intersection types we can't recursively add fields. Basically we want to `ComputedSchema` with `ComputedSchema` everywhere that `Schema` appears.

Reply:
Hmmm. So we're actually trying to create a parallel hierarchy that has the extra metadata at all levels, and instead of creating separate types, we're effectively monkey-patching the existing types. Is that right? Seems a little bit magic to me... How would you implement it if TS did **not** have declaration merging? Maybe the result would be a more clear implementation since it doesn't magically mix types. For example, you could instead create a nested map that is stored only at the top-level of the schema, where the first level of key is the `path` to the relevant inner schema entry.

@timostamm would certainly be a better judge than me as to whether this is appropriate/idiomatic use of declaration merging or if it feels like an abuse. I'm clearly not a good judge because I _like_ the non-magic qualities of Go, so have a bit of an allergy to constructions like this 😛

Reply:
Okay. I'm not a fan, but I'm also not much of a TS developer 😂 

I already approved this PR, so you can merge as is.